pkg/3scale/amp/operator/system_reconciler.go

```diff
@@ -232,31 +232,9 @@ func (r *SystemReconciler) Reconcile() (reconcile.Result, error) {
 		return reconcile.Result{}, err
 	}
 
-	// TODO rethink where to create the system-database secret
-	if r.apiManager.Spec.HighAvailability != nil && r.apiManager.Spec.HighAvailability.Enabled {
-		ha, err := r.highAvailability()
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-
-		err = r.reconcileDatabaseHASecret(ha.SystemDatabaseSecret())
```

Comment:
We could remove the reconcileDatabaseHASecret method itself then too

Reply:
Templates need to create the secret with parameters passed to the template. Method cannot be removed.

Reply:
But the reconcileDatabaseHASecret is an unexported method of the "operator" package (pkg/3scale/amp/operator), it is not used by the templates.

Reply:
sure, I was referring to `ha.SystemDatabaseSecret()`